src/Expr.js

```diff
@@ -169,35 +218,92 @@ var exprToString = function(expr, caller) {
 
   if ('object' in expr) return printObject(expr['object'])
 
-  // Versioned queries/lambdas will have an api_version field.
-  // We want to prevent it from being parsed and displayed as:
-  // Query(ApiVersion("3", "X", Var("X")))
-  var keys = Object.keys(expr).filter(
-    expression => expression !== 'api_version'
-  )
-  var fn = keys[0]
+  if ('select' in expr) {
+    return (
+      'Select(' +
+      exprToString(expr['select']) +
+      ', ' +
+      exprToString(expr['from']) +
+      ')'
+    )
+  }
 
-  // For FQL functions with special formatting concerns, we
-  // use the specialCases object above to define their casing.
-  if (fn in specialCases) fn = specialCases[fn]
+  if ('containsstr' in expr) {
+    return (
+      'ContainsStr(' +
+      exprToString(expr['containsstr']) +
+      ', ' +
+      exprToString(expr['search']) +
+      ')'
+    )
+  }
 
-  fn = fn
-    .split('_')
-    .map(function(str) {
-      return str.charAt(0).toUpperCase() + str.slice(1)
-    })
-    .join('')
+  if ('filter' in expr) {
+    return (
+      'Filter(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['filter']) +
+      ')'
+    )
+  }
+
+  if ('lambda' in expr) {
+    return (
+      'Lambda(' +
+      exprToString(expr['lambda']) +
+      ', ' +
+      exprToString(expr['expr']) +
+      ')'
+    )
+  }
+
+  if ('foreach' in expr) {
+    return (
+      'Foreach(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['foreach']) +
+      ')'
+    )
+  }
+
+  if ('if' in expr) {
+    return (
+      'If(' +
+      exprToString(expr['if']) +
+      ', ' +
+      exprToString(expr['then']) +
+      ', ' +
+      exprToString(expr['else']) +
+      ')'
+    )
+  }
+
+  if ('call' in expr) {
+    return (
+      'Call(' +
+      exprToString(expr['call']) +
+      ', ' +
+      exprToString(expr['arguments']) +
+      ')'
+    )
+  }
+
+  if ('databases' in expr) {
```

Comment:
@marrony I added this to handle the case where tests were failing due to `Databases(null)` being rendered. I'm assuming we'd also want to include `Collections()`, `Documents()`, etc. as well, right?

Reply:
yes, that's correct

Reply:
bear in mind that `Databases()` and its siblings could receive an argument, the scope, so you should check if `expr[databases]` is not null and render it too like `Databases(Database('child-db'))`